creme/neighbors/knn_regressor.py

```diff
@@ -0,0 +1,158 @@
+    aggregating the values of the closest n_neighbors stored-samples with
+    respect to a query sample.
+
+    Parameters
+    ----------
+    n_neighbors : int (default=5)
+        The number of nearest neighbors to search for.
+
+    max_window_size : int (default=1000)
+        The maximum size of the window storing the last observed samples.
+
+    leaf_size : int (default=30)
+        scipy.spatial.cKDTree parameter. The maximum number of samples that
+        can be stored in one leaf node, which determines from which point
+        the algorithm will switch for a brute-force approach. The bigger
+        this number the faster the tree construction time, but the slower
+        the query time will be.
+
+    p : float (default=2)
+        p-norm value for the Minkowski metric. When `p=1`, this corresponds to the
+        Manhattan distance, while `p=2` corresponds to the Euclidean distance. Valid
+        values are in the interval $[1, +\infty)$
+
+    aggregation_method : str (default='mean')
+            | The method to aggregate the target values of neighbors.
+            | 'mean'
+            | 'median'
+            | 'weighted_mean'
+
+    Notes
+    -----
+    This estimator is not optimal for a mixture of categorical and numerical
+    features. This implementation treats all features from a given stream as
+    numerical.
+
+    Examples
+    --------
+    >>> from creme import datasets
+    >>> from creme import evaluate
+    >>> from creme import metrics
+    >>> from creme import neighbors
+    >>> from creme import preprocessing
+
+    >>> dataset = datasets.TrumpApproval()
+
+    >>> model = (
+    ...  preprocessing.StandardScaler() |
+    ...  neighbors.KNNRegressor(max_window_size=50)
+    ... )
+
+    >>> metric = metrics.MAE()
+
+    >>> evaluate.progressive_val_score(dataset, model, metric)
+    # MAE: 0.441308
+
+    """
+
+    _MEAN = 'mean'
+    _MEDIAN = 'median'
+    _WEIGHTED_MEAN = 'weighted_mean'
+
+    def __init__(self, n_neighbors: int = 5, max_window_size: int = 1000, leaf_size: int = 30,
+                 p: float = 2, aggregation_method: str = 'mean'):
+
+        super().__init__(n_neighbors=n_neighbors,
+                         max_window_size=max_window_size,
+                         leaf_size=leaf_size,
+                         p=p)
+        if aggregation_method not in {self._MEAN, self._MEDIAN, self._WEIGHTED_MEAN}:
+            raise ValueError('Invalid aggregation_method: {}.\n'
+                             'Valid options are: {}'.format(aggregation_method,
+                                                            {self._MEAN, self._MEDIAN,
+                                                             self.f_WEIGHTED_MEAN}))
+        self.aggregation_method = aggregation_method
+
+    def learn_one(self, x: dict, y: base.typing.RegTarget) -> 'Regressor':
+        """Fits to a set of features ``x`` and a real-valued target ``y``.
```

Comment:
You can write this as ``x`` and ``y``, no need for two back ticks.

Reply:
Old habits die hard :P
I will fix that!

Reply:
Fixed in [1a91c21](https://github.com/creme-ml/creme/pull/323/commits/1a91c2109076c5b00b947242115ed4f346681c42).